File: 12_Strings_MakingAnagrams.py
'''
https://www.hackerrank.com/challenges/ctci-making-anagrams
'''

import math

def makeAnagram(a, b):

    char_arr = [0] * 26

    for char in a:
        index = ord(char) - ord('a')
        char_arr[index] += 1
    for char in b:
        index = ord(char) - ord('a')
        char_arr[index] -= 1

    result = 0
    for e in char_arr:
        result += int(math.fabs(e))

    return result



a = "abc"
b = "cde"

print(makeAnagram(a, b))

File: 12_Strings_MakingAnagrams.cpp
//
// Created by jan on 10.01.20.
// https://www.hackerrank.com/challenges/ctci-making-anagrams
//

#include <bits/stdc++.h>

using namespace std;

// Complete the makeAnagram function below.
int makeAnagram(string a, string b) {
    vector <int> char_arr (26, 0);
    double count = 0;
    for (int i : a){
        char_arr[i - 'a'] += 1;
    }
    for (int i : b){
        char_arr[i - 'a'] -= 1;
    }
    for (auto val : char_arr) { count += abs(val); }
    return count;
}